### ISSUE TITLE
stats/timing results

### ISSUE THREAD

**Issue by ditchfieldcaleb** [madMAx43v3r/chia-plotter#7]
figured it might be helpful to have some info on how this runs on other machines. here's my results so far - I'll add to this as I actually complete threads, for now this is just the results of phase 1 table 1.

machine: i9-10850k @ 5.0ghz, 64GB RAM
os: ubuntu 20.04 via wsl2 on win10
working directory (1 & 2): Samsung 980 Pro 2TB nvme m.2 ssd

phase 1 table 1 timings - 1 process

threads=64  buckets=6 table1=74  seconds
threads=18  buckets=6 table1=86  seconds

threads=64  buckets=7 table1=75  seconds
threads=36  buckets=7 table1=76  seconds
threads=20  buckets-7 table1=84  seconds
threads=18  buckets=7 table1=88  seconds
threads=14  buckets=7 table1=97  seconds
threads=10  buckets=7 table1=108 seconds
threads=6    buckets=7 table1=120 seconds

threads=64  buckets=8 table1=81  seconds
threads=18  buckets=8 table1=89  seconds

threads=18  buckets=9 table1=92  seconds

**Comment by ditchfieldcaleb** [comment]
interestingly, I couldn't get total cpu usage above about 60% at any given time; I'm assuming the iops/bandwidth of my 980 Pro 2TB is the bottleneck here.

**Comment by madMAx43v3r** [comment]
Table 1 is a bad benchmark, it's quite a different load than the other ones. Much more IO load, than CPU.

**Comment by madMAx43v3r** [comment]
`[P1] Table 1 took 21.0467 sec`
That's when using RAM as `tmp_dir2` and with 16 threads. You're hitting the write speed limit of your SSD big time on Table 1.

**Comment by ditchfieldcaleb** [comment]
> `[P1] Table 1 took 21.0467 sec`
> That's when using RAM as `tmp_dir2` and with 16 threads. You're hitting the write speed limit of your SSD big time on Table 1.

Makes sense - ordered a 128GB kit which should hopefully drastically increase Phase 1 times. Should have some full plot stats here within an hour or two.

**Comment by fermosan** [comment]
This is a run with other 19 plots running in parallel (from original chiapos). 
2 x Intel(R) Xeon(R) CPU E5-2683 v3 @ 2.00GHz 
All plots share the same SSDs divided equally to 4 chunks. 
The system is pretty much **at its limit** at this point.

Number of threads: 8
Number of sort buckets: 2^8 (256)
[P1] Table 1 took 65.5273 sec
[P1] Table 2 took 407.66 sec, found 4295006334 matches
[P1] Lost 38078 matches due to 32-bit overflow.
[P1] Table 3 took 600.319 sec, found 4295069227 matches
[P1] Lost 101461 matches due to 32-bit overflow.
[P1] Table 4 took 688.109 sec, found 4295005288 matches
[P1] Lost 38169 matches due to 32-bit overflow.
[P1] Table 5 took 564.777 sec, found 4294989698 matches
[P1] Lost 21846 matches due to 32-bit overflow.
[P1] Table 6 took 580.173 sec, found 4295038169 matches
[P1] Lost 70663 matches due to 32-bit overflow.
[P1] Table 7 took 454.202 sec, found 4295033899 matches
Phase 1 took 3360.9 sec
[P2] max_table_size = 4295069227
[P2] Table 7 scan took 94.6327 sec
[P2] Table 7 rewrite took 173.6 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 99.7761 sec
[P2] Table 6 rewrite took 191.597 sec, dropped 581347959 entries (13.5353 %)
[P2] Table 5 scan took 91.5333 sec
[P2] Table 5 rewrite took 172.827 sec, dropped 761971771 entries (17.7409 %)
[P2] Table 4 scan took 93.1548 sec
[P2] Table 4 rewrite took 164.58 sec, dropped 828850768 entries (19.298 %)
[P2] Table 3 scan took 92.2078 sec
[P2] Table 3 rewrite took 168.822 sec, dropped 855131659 entries (19.9096 %)
[P2] Table 2 scan took 96.7399 sec
[P2] Table 2 rewrite took 183.811 sec, dropped 865617481 entries (20.154 %)
Phase 2 took 1640.22 sec
Wrote plot header with 140 bytes
[P3-1] Table 2 took 195.069 sec, wrote 3429388853 right entries
[P3-2] Table 2 took 192.468 sec, wrote 3429388853 left entries, 3429388853 final
[P3-1] Table 3 took 200.344 sec, wrote 3439937568 right entries
[P3-2] Table 3 took 191.307 sec, wrote 3439937568 left entries, 3439937568 final
[P3-1] Table 4 took 176.267 sec, wrote 3466154520 right entries
[P3-2] Table 4 took 154.022 sec, wrote 3466154520 left entries, 3466154520 final
[P3-1] Table 5 took 176.593 sec, wrote 3533017927 right entries
[P3-2] Table 5 took 178.232 sec, wrote 3533017927 left entries, 3533017927 final
[P3-1] Table 6 took 224.089 sec, wrote 3713690210 right entries
[P3-2] Table 6 took 217.529 sec, wrote 3713690210 left entries, 3713690210 final
[P3-1] Table 7 took 249.637 sec, wrote 4295033899 right entries
[P3-2] Lost 66603 entries due to 32-bit overflow.
[P3-2] Table 7 took 257.999 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 2417.76 sec, wrote 21877156374 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 157.654 sec, final plot size is 108835549588 bytes
Total plot creation time was 7576.64 sec

Let's check our mad plot now 🤔 

ProofOfSpace check -f mad.plot 200
Total success: 185/200, 92.5% 🥳 🎆 🍾

**Comment by mist0706** [comment]
So it's not done yet but i think we're looking at a new world record within the hour

> Plot Name: plot-k32-2021-06-07-06-40-28dd987f6b2ec8445991a66fe85dd917fbabe41bbee6d1c3556f2db42be7ec2c
> [P1] Table 1 took 9.14506 sec
> [P1] Table 2 took 119.529 sec, found 4294962725 matches
> [P1] Table 3 took 135.206 sec, found 4294979253 matches

**Comment by Damme** [comment]
Haha so much for 'Maximum 10-20% improvement'...
I wish someone could make a chia daemon without python too. I don't have the time sadly.

**Comment by madMAx43v3r** [comment]
That's a fast machine. Also I found checking with 1000 iterations usually yields above 100%.

**Comment by VertiHydro** [comment]
@mist0706 what are your system specs out of interest?

**Comment by mist0706** [comment]
![image](https://user-images.githubusercontent.com/10143260/120974052-f9223d00-c76f-11eb-8f15-09610575d954.png)
5950x with 2666mhz ram

**Comment by 69chargr** [comment]
Ubuntu 20.04
5900x all core OC to 4.65
2x 2TB Inland Platinum RAID0
64GB 3200

Plot Name: plot-k32-2021-06-08-22-39-8747d228fe630e529388ee82e25daea5a29f6e5b362126b938c295507611df94
[P1] Table 1 took 24.8096 sec
[P1] Table 2 took 94.9047 sec, found 4295005556 matches
[P1] Lost 38245 matches due to 32-bit overflow.
[P1] Table 3 took 150.257 sec, found 4295028207 matches
[P1] Lost 60857 matches due to 32-bit overflow.
[P1] Table 4 took 200.829 sec, found 4295013337 matches
[P1] Lost 45380 matches due to 32-bit overflow.
[P1] Table 5 took 192.758 sec, found 4295087110 matches
[P1] Lost 120375 matches due to 32-bit overflow.
[P1] Table 6 took 171.707 sec, found 4295036538 matches
[P1] Lost 69557 matches due to 32-bit overflow.
[P1] Table 7 took 120.433 sec, found 4294961094 matches
Phase 1 took 956.058 sec
[P2] max_table_size = 4295087110
[P2] Table 7 scan took 10.9199 sec
[P2] Table 7 rewrite took 46.4142 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 28.6829 sec
[P2] Table 6 rewrite took 53.0487 sec, dropped 581337606 entries (13.5351 %)
[P2] Table 5 scan took 26.4871 sec
[P2] Table 5 rewrite took 49.8847 sec, dropped 762038528 entries (17.7421 %)
[P2] Table 4 scan took 28.3256 sec
[P2] Table 4 rewrite took 48.7869 sec, dropped 828877948 entries (19.2986 %)
[P2] Table 3 scan took 25.4886 sec
[P2] Table 3 rewrite took 48.9579 sec, dropped 855106230 entries (19.9092 %)
[P2] Table 2 scan took 24.9134 sec
[P2] Table 2 rewrite took 48.8068 sec, dropped 865575282 entries (20.1531 %)
Phase 2 took 450.005 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 47.5261 sec, wrote 3429430274 right entries
[P3-2] Table 2 took 31.5062 sec, wrote 3429430274 left entries, 3429430274 final
[P3-1] Table 3 took 51.2182 sec, wrote 3439921977 right entries
[P3-2] Table 3 took 34.6017 sec, wrote 3439921977 left entries, 3439921977 final
[P3-1] Table 4 took 54.3665 sec, wrote 3466135389 right entries
[P3-2] Table 4 took 32.8395 sec, wrote 3466135389 left entries, 3466135389 final
[P3-1] Table 5 took 52.9282 sec, wrote 3533048582 right entries
[P3-2] Table 5 took 34.0876 sec, wrote 3533048582 left entries, 3533048582 final
[P3-1] Table 6 took 55.1114 sec, wrote 3713698932 right entries
[P3-2] Table 6 took 35.2223 sec, wrote 3713698932 left entries, 3713698932 final
[P3-1] Table 7 took 65.1647 sec, wrote 4294961094 right entries
[P3-2] Table 7 took 51.6132 sec, wrote 4294961094 left entries, 4294961094 final
Phase 3 took 548.402 sec, wrote 21877196248 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 35.8864 sec, final plot size is 108835684962 bytes
Total plot creation time was 1990.39 sec

Proofs 416 / 400, 1.04

Mother of God....

**Comment by 69chargr** [comment]
Second plot was faster. I was using the PC during the first plot.

Plot Name: plot-k32-2021-06-09-00-09-d24436b4bd4f0871b26a68eafb3fa60769a1d3d35f5e506028b34ae34047cdd0
[P1] Table 1 took 12.2591 sec
[P1] Table 2 took 96.7521 sec, found 4294950613 matches
[P1] Table 3 took 154.236 sec, found 4294871360 matches
[P1] Table 4 took 193.88 sec, found 4294970466 matches
[P1] Lost 2857 matches due to 32-bit overflow.
[P1] Table 5 took 190.932 sec, found 4294968294 matches
[P1] Lost 745 matches due to 32-bit overflow.
[P1] Table 6 took 177.279 sec, found 4294993872 matches
[P1] Lost 26367 matches due to 32-bit overflow.
[P1] Table 7 took 122.585 sec, found 4295042485 matches
Phase 1 took 947.996 sec
[P2] max_table_size = 4295042485
[P2] Table 7 scan took 13.0649 sec
[P2] Table 7 rewrite took 37.295 sec, dropped 0 entries (0 %)
[P2] Table 6 scan took 28.7185 sec
[P2] Table 6 rewrite took 52.859 sec, dropped 581269867 entries (13.5337 %)
[P2] Table 5 scan took 26.8092 sec
[P2] Table 5 rewrite took 50.0234 sec, dropped 761953248 entries (17.7406 %)
[P2] Table 4 scan took 25.6456 sec
[P2] Table 4 rewrite took 49.4492 sec, dropped 828870553 entries (19.2986 %)
[P2] Table 3 scan took 25.3471 sec
[P2] Table 3 rewrite took 49.0754 sec, dropped 855036472 entries (19.9083 %)
[P2] Table 2 scan took 26.2195 sec
[P2] Table 2 rewrite took 48.8021 sec, dropped 865574957 entries (20.1533 %)
Phase 2 took 442.474 sec
Wrote plot header with 268 bytes
[P3-1] Table 2 took 44.9643 sec, wrote 3429375656 right entries
[P3-2] Table 2 took 32.8737 sec, wrote 3429375656 left entries, 3429375656 final
[P3-1] Table 3 took 50.5931 sec, wrote 3439834888 right entries
[P3-2] Table 3 took 33.6685 sec, wrote 3439834888 left entries, 3439834888 final
[P3-1] Table 4 took 51.9865 sec, wrote 3466099913 right entries
[P3-2] Table 4 took 33.1702 sec, wrote 3466099913 left entries, 3466099913 final
[P3-1] Table 5 took 52.1563 sec, wrote 3533015046 right entries
[P3-2] Table 5 took 34.0539 sec, wrote 3533015046 left entries, 3533015046 final
[P3-1] Table 6 took 55.4397 sec, wrote 3713724005 right entries
[P3-2] Table 6 took 34.3975 sec, wrote 3713724005 left entries, 3713724005 final
[P3-1] Table 7 took 62.5906 sec, wrote 4295042485 right entries
[P3-2] Lost 75189 entries due to 32-bit overflow.
[P3-2] Table 7 took 53.3106 sec, wrote 4294967296 left entries, 4294967296 final
Phase 3 took 541.674 sec, wrote 21877016804 entries to final plot
[P4] Starting to write C1 and C3 tables
[P4] Finished writing C1 and C3 tables
[P4] Writing C2 table
[P4] Finished writing C2 table
Phase 4 took 36.839 sec, final plot size is 108834963207 bytes
Total plot creation time was 1969.03 sec

**Comment by VertiHydro** [comment]
Christ alive...

**Comment by VertiHydro** [comment]
No ramdisk and 30 min plot...

**Comment by 69chargr** [comment]
Funny thing is using modified chiapos I was getting 43 plots/day using Swar. If I divide seconds in a day 86,400 by 1969.03 I get..... 43 :-)
So are we really gaining anything here? lol

**Comment by VertiHydro** [comment]
Yes, but now try running multiple instances of this plotter in parallel - because you should be able to fit 12 plots at once on your SSDs, and maybe (I haven't tested parallel yet) it will do 12 30-min plots (probably more like an hour because available CPU will go way down if you divy it all up). Therefore, 12 plots at 1 hr each = 288 plots/day. I doubt it will really work that well, but only way to know for sure is to test!

**Comment by VertiHydro** [comment]
So yeah, I don't know how well parallel will work - I'm going to test it in an hour or so. Will report back.

**Comment by 69chargr** [comment]
Yep, trying 2 now.

**Comment by 69chargr** [comment]
So far it's just doubling the time on each part of the phases. So yeah, no massive breakthrough there.

**Comment by VertiHydro** [comment]
Yeah fair enough. About what I expected.

**Comment by madMAx43v3r** [comment]
You have to overlap phase 1 and 3, when running two parallel.

**Comment by VertiHydro** [comment]
Interesting. I'll give it a shot.

**Comment by madMAx43v3r** [comment]
Also this plotter doesn't reduce total amount of CPU work needed per plot, yet. For now it's for people with SSD bottlenecks. And generally to save money and power draw by using RAM instead of NVMe or large disk arrays.

**Comment by 69chargr** [comment]
Did more testing. Yeah I bottlenecked my NVMe raid 0 so hard trying to overlap phases 1 & 3 on parallel processes that my CPU usage started to plummet. So for me at least it has zero advantage over pechy combined + Swar.

**Comment by pwagma** [comment]
Do you know where logs get saved? I would like to check the output
